build.gradle.kts

```diff
@@ -11,6 +11,41 @@ repositories {
     mavenCentral()
 }
 
+val kotlinVersion = "1.8.10"
+val exposedVersion = "0.37.3"
+val lsp4jVersion = "0.15.0"
+
+// Constrain the dependencies that we use to these specific versions.
+// When these dependencies are defined in subsequent builds, they will need
+// to match the versions defined here.
+subprojects {
+    apply(plugin = "java-library") // needed to register the api() constraint
```

Comment:
I think using `subprojects` and imperative plugin applications is the legacy way of doing this and[generally discouraged](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#sec:convention_plugins_vs_cross_configuration) in favor of [version catalogs and build platforms](https://docs.gradle.org/current/userguide/platforms.html)

Reply:
In my notes, I've got a version catalog as the alternative to a BOM for modern Gradle. I went with the BOM not too long ago because of the syntax but reading the docs here, I like how we can define libraries in a settings file, such as:

```
dependencyResolutionManagement {
    versionCatalogs {
        create("libs") {
            version("groovy", "3.0.5")
            version("checkstyle", "8.37")
            library("groovy-core", "org.codehaus.groovy", "groovy").versionRef("groovy")
            library("groovy-json", "org.codehaus.groovy", "groovy-json").versionRef("groovy")
            library("groovy-nio", "org.codehaus.groovy", "groovy-nio").versionRef("groovy")
            library("commons-lang3", "org.apache.commons", "commons-lang3").version {
                strictly("[3.8, 4.0[")
                prefer("3.9")
            }
        }
    }
}
```

And then use them elsewhere by:

```
dependencies {
    implementation(libs.groovy.core)
    implementation(libs.groovy.json)
    implementation(libs.groovy.nio)
}
```

To me, the catalog written this way is clearer than both the TOML syntax and the BOM that's already present. What do you think @themkat and @fwcd ? Perhaps this would be a better way forward.